etc/rsync/rsyncd.conf.j2

```diff
@@ -0,0 +1,61 @@
+log file = /var/log/rsyncd.log
+pid file = /var/run/rsyncd.pid
+lock file = /var/lock/rsyncd
+uid=root
+gid=root
+{% if 'logserver-slave' in group_names %}
+
+[openlab_logserver]
+comment = sync zuul logs
+path = /srv/static/logs/
+use chroot=yes
+read only = no
+max connections=10
+auth users = root
+secrets file = /etc/rsyncd/rsyncd.pass
+hosts allow={{ hostvars[groups['logserver-master'][0]].ansible_host }}
+ignore errors = yes
+timeout = 600
+{% endif %}
+{% if 'zuul-scheduler-slave' in group_names %}
+
+[openlab_zuul_cfg]
+comment = sync zuul main.yaml
+path = /etc/zuul/
+use chroot=yes
+read only = no
+max connections=10
+auth users = root
+secrets file = /etc/rsyncd/rsyncd.pass
+hosts allow={{ hostvars[groups['zuul-scheduler-master'][0]].ansible_host }}
+ignore errors = yes
+timeout = 600
+{% endif %}
+{% if 'nodepool-launcher-slave' in group_names %}
+
+[openlab_clouds_cfg]
+comment = sync clouds.yaml
+path = /etc/openstack/
+use chroot=yes
+read only = no
+max connections=10
+auth users = root
+secrets file = /etc/rsyncd/rsyncd.pass
+hosts allow={{ hostvars[groups['nodepool-launcher-master'][0]].ansible_host }}
+ignore errors = yes
+timeout = 600
+{% endif %}
+{% if 'nodepool-launcher-slave' in group_names %}
```

Comment:
So the condition is same as line 34? why we need two ?

Reply:
it is could be in one condition, but I think it is better to handle separately to make things flexible if we need to change the deployment in the future. you can also check the playbook.

Reply:
you mean clouds.yaml and nodepool.yaml will be hosted in two nodes one day?

Reply:
hmm.. yes, you are right, my fault. :)

Reply:
> Yes, I think we can merged this two `if ... endif` block

yes, will do :)